ironic/drivers/modules/drac/inspect.py

```diff
@@ -35,16 +42,154 @@
+                    # Get MAC address of the given nic_id
+                    mac_response = requests.get('%s%s%s' % (
+                        redfish_ip, _MAC_URL, nic_id), auth=(
+                        redfish_username, redfish_password), verify=False)
+                    mac_address = json.loads(mac_response.__dict__.get(
+                        '_content')).get('MACAddress')
+                    pxe_dev_nics.append(mac_address)
+
+        elif system.boot.mode == 'bios':
+            url = '%s%s' % (redfish_ip, _JOB_ID_URL)
+            job_response = requests.post(url, data=json.dumps(
+                _PAYLOAD), headers=_HEADERS, verify=False, auth=(
+                redfish_username, redfish_password))
+
+            if job_response.status_code != _JOB_RESPONSE_CODE:
+                error = (_('FAIL, status code not 202, code is: %(code), '
+                           'Error details: %(err)'),
+                         {'code': job_response.status_code,
+                          'err': job_response.__dict__})
+                raise exception.HardwareInspectionFailure(error=error)
+
+            job_id = job_response.__dict__['headers']['Location']
+            try:
+                job_id = re.search('JID_.+', job_id).group()
+            except drac_exceptions.BaseClientException as exc:
+                LOG.error('FAIL: detailed error message: %(err)',
+                          {'err': job_response.__dict__['_content']})
+                raise exception.HardwareInspectionFailure(error=exc)
+
+            start_time = datetime.now()
+
+            while True:
+                current_time = (datetime.now() - start_time)
+                xml_data = requests.get('%s%s%s' % (
+                    redfish_ip, _XML_DATA_URL, job_id), headers=_HEADERS,
+                    auth=(redfish_username, redfish_password), verify=False)
+
+                if _SYSTEM_CONFIG_TAG in str(xml_data.__dict__):
+                    xml_string = xml_data.__dict__.get('_content')
+                    root = ET.fromstring(xml_string)
+                    for child in root:
+                        nic_id = [child.attrib.get('FQDD') for ch in child
+                                  if ch.text == 'PXE']
+                        if nic_id:
+                            # Get MAC address of the given nic_id
+                            mac_response = requests.get('%s%s%s' % (
+                                redfish_ip, _MAC_URL, nic_id[0]), auth=(
+                                redfish_username, redfish_password),
+                                verify=False)
+                            mac_address = json.loads(mac_response.__dict__.get(
+                                '_content')).get('MACAddress')
+                            pxe_dev_nics.append(mac_address)
+                    return pxe_dev_nics
+                elif current_time.seconds >= _TIMEOUT_SECONDS:
+                    error = (_('FAIL, Timeout of %(second) seconds has been '
```

Comment:
Can you change the error message to something like:
The job to export the system configuration did not complete within xxx seconds.

Reply:
> Can you change the error message to something like:
> The job to export the system configuration did not complete within xxx seconds.

Done